pyiron_atomistics/atomistics/master/phonopy.py

```diff
@@ -406,27 +402,32 @@ def dynamical_matrix_at_q(self, q):
         """
         return np.real_if_close(self.phonopy.get_dynamical_matrix_at_q(q))
 
-    def plot_dos(self, ax=None, *args, **qwargs):
+    @deprecate(ax="Use axis instead")
+    def plot_dos(self, ax=None, *args, axis=None, **kwargs):
         """
+        Plot the DOS.
 
         Args:
-            *args:
-            ax:
-            **qwargs:
+            axis (optional): matplotlib axis to use, if None create a new one
+            ax: deprecated alias for axis
+            *args: passed to `axis.plot`
+            **kwargs: passed to `axis.plot`
 
         Returns:
-
+            matplotlib.axes._subplots.AxesSubplot: axis with the plot
         """
         try:
             import pylab as plt
         except ImportError:
             import matplotlib.pyplot as plt
-        if ax is None:
-            fig, ax = plt.subplots(1, 1)
-        ax.plot(self["output/dos_energies"], self["output/dos_total"], *args, **qwargs)
-        ax.set_xlabel("Frequency [THz]")
-        ax.set_ylabel("DOS")
-        ax.set_title("Phonon DOS vs Energy")
+        if ax is not None and axis is None:
```

Comment:
I think you can one line this like _, ax = ply.subplots() if axis is None else None, axis. If you hate it don't bother but else a 4:1 line ratio is not too shabby

Reply:
I guess this wouldn't work when `ax` is defined but not `axis`.  I don't like to layer inline if/else to deeply, so I'd rather forgo it.

Reply:
Ah, you're right. Can still one-line the two clauses independently if you want

```python
axis = ax if ax is not None and axis is None else axis
_, axis = plt.subplots() if axis is None else _, axis
```

The first clause needs to be expanded if you want to explicitly throw an error when both the old and new kwargs are used though.